x-pack/plugins/security_solution/server/lib/siem_migrations/rules/data/rule_migrations_data_rules_client.ts

```diff
@@ -117,65 +109,63 @@ export class RuleMigrationsDataClient {
   }
 
   /** Updates one rule migration with the provided data and sets the status to `completed` */
-  async saveFinished({ _id, _index, ...ruleMigration }: StoredRuleMigration): Promise<void> {
+  async saveCompleted({ _id, ...ruleMigration }: StoredRuleMigration): Promise<void> {
+    const index = await this.getIndexName();
     const doc = {
       ...ruleMigration,
       status: SiemMigrationStatus.COMPLETED,
-      updated_by: this.currentUser.username,
+      updated_by: this.username,
       updated_at: new Date().toISOString(),
     };
-    await this.esClient
-      .update({ index: _index, id: _id, doc, refresh: 'wait_for' })
-      .catch((error) => {
-        this.logger.error(`Error updating rule migration status to completed: ${error.message}`);
-        throw error;
-      });
+    await this.esClient.update({ index, id: _id, doc, refresh: 'wait_for' }).catch((error) => {
+      this.logger.error(`Error updating rule migration status to completed: ${error.message}`);
+      throw error;
+    });
   }
 
   /** Updates one rule migration with the provided data and sets the status to `failed` */
-  async saveError({ _id, _index, ...ruleMigration }: StoredRuleMigration): Promise<void> {
+  async saveError({ _id, ...ruleMigration }: StoredRuleMigration): Promise<void> {
+    const index = await this.getIndexName();
     const doc = {
       ...ruleMigration,
       status: SiemMigrationStatus.FAILED,
-      updated_by: this.currentUser.username,
+      updated_by: this.username,
       updated_at: new Date().toISOString(),
     };
-    await this.esClient
-      .update({ index: _index, id: _id, doc, refresh: 'wait_for' })
-      .catch((error) => {
-        this.logger.error(`Error updating rule migration status to completed: ${error.message}`);
-        throw error;
-      });
+    await this.esClient.update({ index, id: _id, doc, refresh: 'wait_for' }).catch((error) => {
+      this.logger.error(`Error updating rule migration status to failed: ${error.message}`);
```

Comment:
side note: it would be great to track telemetry on the UI on our success and failure rates and retry count as well